test/metabase/api/search_test.clj

```diff
@@ -718,13 +722,17 @@
                                :name     "segment count test 2"}
      Segment   _              {:table_id table-id
                                :name     "segment count test 3"}]
-    (with-redefs [premium-features/sandboxed-or-impersonated-user? (constantly false)]
+    (mt/with-current-user (mt/user->id :crowberto)
       (toucan2.execute/with-call-count [call-count]
-        (#'api.search/search (#'api.search/search-context "count test" nil nil nil 100 0))
+        (#'api.search/search {:search-string      "count test"
+                              :archived?          false
+                              :models             search-config/all-models
+                              :current-user-perms #{"/"}
+                              :limit-int          100})
         ;; the call count number here are expected to change if we change the search api
         ;; we have this test here just to keep tracks this number to remind us to put effort
         ;; into keep this number as low as we can
-        (is (= 7 (call-count)))))))
+        (is (= 10 (call-count)))))))
```

Comment:
Actually, there is an improvement here. The previous number `7` is wrong because the `query-model-set` returns a lazy seq and the previous test didn't realize it, so `7` was not a real number from the API. The correct number was 19.